File: autodroid-app/app/src/main/java/com/autodroid/proxy/ReportDetailActivity.java
package com.autodroid.proxy;

import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import com.autodroid.proxy.viewmodel.AppViewModel;
import com.google.gson.Gson;
import com.google.gson.JsonObject;

import java.util.Map;

public class ReportDetailActivity extends AppCompatActivity {

    private static final String TAG = "ReportDetailActivity";
    
    private TextView reportIdTextView;
    private TextView workflowNameTextView;
    private TextView statusTextView;
    private TextView startTimeTextView;
    private TextView durationTextView;
    private TextView stepsTextView;
    
    private AppViewModel viewModel;
    private Gson gson;
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_report_detail);
        
        // Initialize Gson
        gson = new Gson();
        
        // Initialize UI components
        reportIdTextView = findViewById(R.id.report_id);
        workflowNameTextView = findViewById(R.id.report_workflow_name);
        statusTextView = findViewById(R.id.report_status);
        startTimeTextView = findViewById(R.id.report_start_time);
        durationTextView = findViewById(R.id.report_duration);
        stepsTextView = findViewById(R.id.report_steps);
        
        // Initialize ViewModel
        viewModel = new ViewModelProvider(this).get(AppViewModel.class);
        
        // Get report ID from intent
        String reportId = getIntent().getStringExtra("report_id");
        
        // Observe selected report changes
        viewModel.getSelectedReport().observe(this, report -> {
            if (report != null) {
                updateReportUI(report);
            }
        });
        
        // Load report details (in a real app, this would fetch from server or database)
        if (reportId != null) {
            loadReportDetails(reportId);
        }
    }
    
    private void loadReportDetails(String reportId) {
        // In a real app, this would fetch the report details from the server
        // For now, we'll just use a mock report
        Map<String, Object> mockReport = Map.of(
            "id", reportId,
            "workflow_name", "Sample Workflow",
            "status", "success",
            "start_time", "2024-01-01 12:00:00",
            "duration", "30.5",
            "steps", "Step 1: Launch App - Success\nStep 2: Click Button - Success\nStep 3: Verify Result - Success"
        );
        
        viewModel.setSelectedReport(mockReport);
    }
    
    private void updateReportUI(Map<String, Object> report) {
        reportIdTextView.setText("Report ID: " + report.getOrDefault("id", "N/A"));
        workflowNameTextView.setText("Workflow: " + report.getOrDefault("workflow_name", "N/A"));
        statusTextView.setText("Status: " + report.getOrDefault("status", "N/A"));
        startTimeTextView.setText("Start Time: " + report.getOrDefault("start_time", "N/A"));
        durationTextView.setText("Duration: " + report.getOrDefault("duration", "N/A") + " seconds");
        stepsTextView.setText("Steps:\n" + report.getOrDefault("steps", "N/A"));
        
        // Set status text color based on status
        String status = (String) report.getOrDefault("status", "");
        if (status.equals("success")) {
            statusTextView.setTextColor(getResources().getColor(android.R.color.holo_green_dark));
        } else if (status.equals("failed")) {
            statusTextView.setTextColor(getResources().getColor(android.R.color.holo_red_dark));
        } else {
            statusTextView.setTextColor(getResources().getColor(android.R.color.black));
        }
    }
}

File: autodroid-app/app/src/main/java/com/autodroid/proxy/ReportDetailActivity.kt
// ReportDetailActivity.kt
package com.autodroid.proxy

import android.os.Bundle
import android.util.Log
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import com.autodroid.proxy.viewmodel.AppViewModel
import com.google.gson.Gson
import java.util.*

class ReportDetailActivity : AppCompatActivity() {
    companion object {
        private const val TAG = "ReportDetailActivity"
    }

    private lateinit var reportIdTextView: TextView
    private lateinit var workflowNameTextView: TextView
    private lateinit var statusTextView: TextView
    private lateinit var startTimeTextView: TextView
    private lateinit var durationTextView: TextView
    private lateinit var stepsTextView: TextView

    private lateinit var viewModel: AppViewModel
    private lateinit var gson: Gson

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_report_detail)

        // Initialize Gson
        gson = Gson()

        // Initialize UI components
        reportIdTextView = findViewById(R.id.report_id)
        workflowNameTextView = findViewById(R.id.report_workflow_name)
        statusTextView = findViewById(R.id.report_status)
        startTimeTextView = findViewById(R.id.report_start_time)
        durationTextView = findViewById(R.id.report_duration)
        stepsTextView = findViewById(R.id.report_steps)

        // Initialize ViewModel
        viewModel = ViewModelProvider(this)[AppViewModel::class.java]

        // Get report ID from intent
        val reportId = intent.getStringExtra("report_id")

        // Observe selected report changes
        viewModel.selectedReport.observe(this) { report ->
            if (report != null) {
                updateReportUI(report as Map<String, Any>)
            }
        }

        // Load report details (in a real app, this would fetch from server or database)
        if (reportId != null) {
            loadReportDetails(reportId)
        }
    }

    private fun loadReportDetails(reportId: String) {
        // In a real app, this would fetch the report details from the server
        // For now, we'll just use a mock report
        val mockReport: MutableMap<String, Any> = HashMap()
        mockReport["id"] = reportId
        mockReport["workflow_name"] = "Sample Workflow"
        mockReport["status"] = "success"
        mockReport["start_time"] = "2024-01-01 12:00:00"
        mockReport["duration"] = "30.5"
        mockReport["steps"] = "Step 1: Launch App - Success\nStep 2: Click Button - Success\nStep 3: Verify Result - Success"

        viewModel.setSelectedReport(mockReport as MutableMap<String?, Any?>?)
    }

    private fun updateReportUI(report: Map<String, Any>) {
        reportIdTextView.text = "Report ID: ${report.getOrDefault("id", "N/A")}"
        workflowNameTextView.text = "Workflow: ${report.getOrDefault("workflow_name", "N/A")}"
        statusTextView.text = "Status: ${report.getOrDefault("status", "N/A")}"
        startTimeTextView.text = "Start Time: ${report.getOrDefault("start_time", "N/A")}"
        durationTextView.text = "Duration: ${report.getOrDefault("duration", "N/A")} seconds"
        stepsTextView.text = "Steps:\n${report.getOrDefault("steps", "N/A")}"

        // Set status text color based on status
        val status = report.getOrDefault("status", "").toString()
        when (status) {
            "success" -> statusTextView.setTextColor(resources.getColor(android.R.color.holo_green_dark))
            "failed" -> statusTextView.setTextColor(resources.getColor(android.R.color.holo_red_dark))
            else -> statusTextView.setTextColor(resources.getColor(android.R.color.black))
        }
    }
}